src/UnigramWordPredictor.java

```diff
@@ -48,10 +49,21 @@ public UnigramWordPredictor(Tokenizer tokenizer) {
    * 
    * @param scanner the Scanner to read the training text from
    */
+
+  //Scanner for readig the text
+  //builds neighborMap by iterating through the tokens.
+  //It identifies the word that follows it and it adds it to the list of that current word
   public void train(Scanner scanner) {
-    List<String> trainingWords = tokenizer.tokenize(scanner);
+    List<String> trainingWords = tokenizer.tokenize(scanner); 
+    this.neighborMap = new HashMap<>();  
```

Comment:
This works well! You can make it a bit simpler by omitting the `this`.
```
neighborMap = new HashMap<>();
```
It does the same thing: for any variables that are not otherwise declared in the local scope, Java assumes they are instance variables and essentially puts an invisible `this.` in front of them.